commit cb90bf04e6193e684969ff2688ec5e4b82849397 (refs/changes/39/2539/2)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2017-09-11T03:19:44+00:00 (2 years, 1 month ago)
    
    MANATEE-357 RST flood protection turns active failures into timeouts when PostgreSQL not running

diff --git a/boot/configure.sh b/boot/configure.sh
index 7460723..b67020c 100755
--- a/boot/configure.sh
+++ b/boot/configure.sh
@@ -1,5 +1,4 @@
 #!/bin/bash
-# -*- mode: shell-script; fill-column: 80; -*-
 #
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
@@ -7,10 +6,19 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2017, Joyent, Inc.
 #
 
-export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
+export PS4
 set -o xtrace
 
+#
+# Disable the protection against RST reflection denial-of-service attacks.
+# In order for system liveliness when PostgreSQL is not running, we need
+# to be able to send a RST for every inbound connection to a closed port.
+# This is only safe because we run Manatee on an isolated network.
+#
+/usr/sbin/ndd -set /dev/tcp tcp_rst_sent_rate_enabled 0
+
 exit 0
